### PR TITLE
[@mantine/dates] - Timerangeinput Component

### PR DESCRIPTION
**Pull request by achmurali**
This is just a intial PR for reviewing, shall write the tests once confirmed

### CODE REVIEW

File: src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.story.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const sizes = MANTINE_SIZES.map((size) => (
+  <TimeRangeInput label={size} size={size} key={size} style={{ marginTop: 30 }} />
+));
+
+function Controlled() {
+  const [value, setValue] = useState([new Date(), new Date()]);
+  const onChange = (val) => {
+    setValue(val);
+  };
+  return <TimeRangeInput value={[value[0], value[1]]} onChange={onChange} label="Controlled" withSeconds />;
```
Comment (rtivital) [inline src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.story.tsx:17]: Just 

```tsx
<TimeRangeInput value={value} onChange={setValue} label="Controlled" withSeconds />
```

No need for code duplication

File: src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.styles.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+export default createMemoStyles({
+  disabled: {
+    opacity: 0.6,
+    cursor: 'not-allowed',
+  },
+
+  wrapper: {
+    paddingTop: '5px',
```
Comment (rtivital) [inline src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.styles.ts:29]: 1. Why wrapper has padding top?
2. When you add px values, set value as number: `paddingTop: 5`, not `paddingTop: '5px'`

File: src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.story.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+storiesOf('@mantine/dates/TimeRangeInput', module)
+  .add('General usage', () => (
+    <TimeRangeInput
+      label="Time Range input"
+      disabled
+      styles={{
+      wrapper: {
```
Comment (rtivital) [inline src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.story.tsx:26]: Please make sure that you have prettier installed in your code editor, there are lots formatting issues that can be automatically fixed

File: src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+              style={_styles.timeInput}
+              size={size}
+              max={59}
+              aria-label={secondsLabel}
+              disabled={disabled}
+              onFocus={() => handleFieldFocus(0)}
+            />
+          )}
+
+          <span tabIndex={-1}>{ labelSeparator }</span>
```
Comment (rtivital) [inline src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.tsx:264]: `tabIndex` not needed, span is not an interactive element

**Review by rtivital — COMMENTED** [review]
Thanks for PR! Works fine but there is to little space between dash and time fields, it should be increased

**Comment by achmurali** [comment]
Have implemented checks to ensure the toTime value is always greater than from.

**Comment by rtivital** [comment]
Not sure about this check in terms of UX
for example, I can see how `23:30 – 00:45` can be a valid time

**Comment by achmurali** [comment]
> Not sure about this check in terms of UX
> for example, I can see how `23:30 – 00:45` can be a valid time

Yeah thought of this, but for that to happen it would only sound reasonable if there was also DD/MM/YYYY along with it. But anyways should I remove it or make it optional via a prop?

**Comment by rtivital** [comment]
Yes, it would be best to have this option via prop

**Comment by achmurali** [comment]
Can't come up with a good name for that prop, any ideas 😓😁

**Comment by achmurali** [comment]
@rtivital, after a bit of pondering I believe this kind of check to keep toTime greater than fromTime requires a lot more checks and unnecessarily introduces complexity 😓  -
- Like, if fromTime is 12:59:59 then the toTime should be clamped to minimum of 13:00:00.
- Similarly, if fromTime is 12:43:59 then the toTime should be clamped to minimum of 12:44:00.

Please, let me know if you want me to implement the above?

**Comment by rtivital** [comment]
Nah, don't waste your time, it'd be better to drop this feature

**Comment by achmurali** [comment]
Hey @rtivital, have updated the component, implemented tests and docs. Please go through and let me know in case of any corrections

**Comment by rtivital** [comment]
Alright, thanks, I'll have a look later today or tomorrow.

Can you also add it to changelog 2.5.0?

**Comment by rtivital** [comment]
Please make sure you are in sync with current next-minor branch, there is already 2.5.0 changelog

**Comment by achmurali** [comment]
> Please make sure you are in sync with current next-minor branch, there is already 2.5.0 changelog

Sorry, My bad!...I branched my branch from master, hence this issue

File: docs/src/docs/changelog/2-5-0.mdx [hunk 5]
```diff
@@ -66,6 +68,12 @@ With [Dialog](https://mantine.dev/core/dialog/) component you can display fixed
 
 <Demo data={ListDemos.icon} demoProps={{ toggle: true }} />
 
+## New @mantine/dates components
+
+[TimeInput](/dates/time-input/) component allows to capture time range from user:
```
Comment (rtivital) [inline docs/src/docs/changelog/2-5-0.mdx:73]: TimeRangeInput, not TimeInput

File: docs/src/docs/changelog/2-5-0.mdx [hunk 6]
```diff
@@ -66,6 +68,12 @@ With [Dialog](https://mantine.dev/core/dialog/) component you can display fixed
 
 <Demo data={ListDemos.icon} demoProps={{ toggle: true }} />
 
+## New @mantine/dates components
```
Comment (rtivital) [inline docs/src/docs/changelog/2-5-0.mdx:71]: Delete this title, no need to separate packages

File: docs/src/docs/dates/time-range-input.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+package: '@mantine/dates'
+title: TimeRangeInput
+order: 1
+slug: /dates/time-range-input/
+description: 'Capture time range from user'
+props: ['TimeRangeInput']
+import: "import { TimeRangeInput } from '@mantine/dates';"
+source: 'mantine-dates/src/components/TimeRangeInput/TimeRangeInput.tsx'
+docs: 'dates/TimeRangeInput.mdx'
+styles: ['TimeRangeInput']
```
Comment (rtivital) [inline docs/src/docs/dates/time-range-input.mdx:12]: Please check that styles api tab works correctly, it does not display anything
Снимок экрана 2021-09-12 в 15 17 56

File: src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.styles.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  MantineSize,
+  getSizeValue,
+} from '@mantine/core';
+
+interface TimeRangeInputStyles {
+  theme: MantineTheme;
+  size: MantineSize;
+}
+
+const inputSizes = {
```
Comment (rtivital) [inline src/mantine-dates/src/components/TimeRangeInput/TimeRangeInput.styles.ts:14]: Reuse variable from TimeInput component, do not create new one

File: src/mantine-dates/src/components/TimeRangeInput/demos/usage.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+const code = `
+import { useState } from 'react';
+import dayjs from 'dayjs';
+import { DateRangePicker } from '@mantine/dates';
+
+function Demo() {
+  const [value, setValue] = useState<[Date, Date]>([new Date(), new Date()]);
+
+  return (
+    <DateRangePicker
```
Comment (rtivital) [inline src/mantine-dates/src/components/TimeRangeInput/demos/usage.tsx:13]: Incorrect demo code

File: src/mantine-dates/src/components/TimeRangeInput/demos/usage.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import { TimeRangeInput } from '../TimeRangeInput';
+
+const code = `
+import { useState } from 'react';
+import dayjs from 'dayjs';
```
Comment (rtivital) [inline src/mantine-dates/src/components/TimeRangeInput/demos/usage.tsx:6]: include only those dependencies which are used in demo, remove dayjs

**Comment by achmurali** [comment]
Gee, forgot to apply my stash after merging the branch...Sorry 😓

**Comment by rtivital** [comment]
Thanks, look good now!

**Comment by rtivital** [comment]
@achmurali I've made some changes after merge, there were some bugs:

- component did not work in uncontrolled mode (you've used `value` instead of `_value` to convert to date)
- you've used `elementRef` twice on hours inputs
- styles api were missing

I've fixed those things, you can review these changes here:
- General bug fixed – https://github.com/mantinedev/mantine/commit/fa3847f513dcc690c74e5a93ae422548736614b7
- How to add styles api (including tests) – https://github.com/mantinedev/mantine/commit/808ab24073154d514d5f1ba8a375447015fb8a96

**Comment by achmurali** [comment]
Thanks @rtivital.

**Pull request merged**